tools/gdb/gdb_scripts/hazard_pointers.py

```diff
@@ -0,0 +1,80 @@
+def find_sessions():
+    original_frame = gdb.selected_frame()
+    sessions = []
+
+    current_prog = gdb.inferiors()[0]
+    for thread in current_prog.threads():
+        thread.switch()
+
+        # Walk up the stack until we find a function that takes session as an argument
+        gdb.newest_frame()
+        cur_frame = gdb.selected_frame()
+        while cur_frame is not None:
+            cur_frame.select()
+            session_arg = gdb.execute("info arg session", to_string=True)
+            if session_arg.startswith("session = 0x"):
+                sessions.append((f"Thread {thread.global_num}", session_arg[10:].strip()))
+                break
+            cur_frame = cur_frame.older()
+
+    # Return back to our original frame from before we called this function
+    original_frame.select()
+    return sessions
+
+# Helper script to dump all active hazard pointers and the sessions that hold them.
+# Usage: `dump_hazard_pointers`
+class DumpHazardPointers(gdb.Command):
+    def __init__(self):
+        super(DumpHazardPointers, self).__init__("dump_hazard_pointers", gdb.COMMAND_DATA)
+
+    def invoke(self, arg, from_tty):
+        for (thread, session_ptr) in find_sessions():
+            print(f"\n{thread}, session ({session_ptr}):")
+            if(session_ptr == "0x0"):
+                continue
+            session = gdb.parse_and_eval(session_ptr).reinterpret_cast(SESSION_IMPL_PTR).dereference()
+            if session['active']:
+                for j in range(0, session['hazards']['inuse']):
+                    hazard_ptr = session['hazards']['arr'][j]
+                    if hazard_ptr['ref'] != 0:
+                        print(f"    {hazard_ptr}")
+
+# Helper script that lists all threads that hold a hazard pointer on the provided ref.
+# Usage: `find_hazard_pointers_for 0xffff8c5792a0`
+class FindHazardPointerOf(gdb.Command):
+    def __init__(self):
+        super(FindHazardPointerOf, self).__init__("find_hazard_pointers_for", gdb.COMMAND_DATA)
+
+    def invoke(self, arg, from_tty):
+        if(len(arg.split()) != 1 or not str(arg).startswith("0x")):
+            print("Example usage: `find_hazard_pointers_for 0xffff8c5792a0``")
+            return 1
+
+        target_ref = str(arg)
+        print(f"Threads, sessions with a hazard pointer on {target_ref}:")
+        for (thread, session_ptr) in find_sessions():
+            if(session_ptr == "0x0"):
+                continue
+            session = gdb.parse_and_eval(session_ptr).reinterpret_cast(SESSION_IMPL_PTR).dereference()
+            if session['active']:
+                for j in range(0, session['hazards']['inuse']):
+                    hazard_ptr = session['hazards']['arr'][j]
+                    if str(hazard_ptr['ref']) == str(target_ref):
+                        print(f"    {thread}, Session {session_ptr}")
+                        print(f"        {hazard_ptr}")
+
+# Register scripts with gdb when `source` is called.
+DumpHazardPointers()
```

Comment:
While we don't have a style for this, it is inconsistent with other scripts which use snake_case or their registered functions. Unless you want a fun 10/10 style discussion I'd suggest being in line with it.

Reply:
Updated in [85bdb20](https://github.com/wiredtiger/wiredtiger/pull/9801/commits/85bdb20849cbd8afde6506bf1d480f568de9340e)